test/distributed/_shard/sharded_tensor/ops/test_linear.py

```diff
@@ -70,6 +70,7 @@ def _run_sharded_linear(
         inp = torch.rand(*input_size).cuda(self.rank)
         reshard_spec = copy.deepcopy(spec)
         reshard_spec.dim = 0
+        reshard_spec.placements.sort(key=lambda placement: placement.rank())
```

Comment:
curious why we need to sort the placements here?

Reply:
Oh because we want to send back result of rank 0 to rank 0 because the placements are not always sorted by rank.